inttest/util/k8s.go

```diff
@@ -0,0 +1,114 @@
+package util
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"testing"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/yaml"
+)
+
+func KubectlApply(t *testing.T, kubeconfig string, namespace string, file string) {
+	t.Logf("applying %s", file)
+	out, err := exec.Command("kubectl", "--kubeconfig", kubeconfig, "apply", "-f", file, "-n", namespace).CombinedOutput()
+	if err != nil {
+		t.Logf("output: %s", out)
+		t.Fatalf("failed to apply %s: %s", file, err)
+	}
+	t.Logf("applied %s", file)
+}
+
+func WaitForPodComplete(t *testing.T, kubeconfig string, namespace string, name string, timeout time.Duration) {
+	t.Logf("waiting for pod %s:%s to be succeeded", namespace, name)
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+	err := wait.PollUntilContextCancel(ctx, 2*time.Second, true, func(ctx context.Context) (bool, error) {
+		return isPodSucceeded(kubeconfig, namespace, name)
+	})
+	if err != nil {
+		t.Fatalf("failed to wait for pod %s:%s: %s", namespace, name, err)
+	}
+	t.Logf("pod %s:%s is succeeded", namespace, name)
+}
+
+func isPodSucceeded(kubeconfig string, namespace string, name string) (bool, error) {
+	cmd := exec.Command(
+		"kubectl", "--kubeconfig", kubeconfig, "get", "pod", name, "-n", namespace,
+		"-o", "jsonpath={.status.phase}",
+	)
+	out, err := cmd.Output()
+	if err != nil {
+		return false, err
+	}
+	if string(out) == "Succeeded" {
+		return true, nil
+	}
+	return false, nil
+}
+
+func WaitForDeployment(t *testing.T, kubeconfig string, namespace string, name string, replicas int, timeout time.Duration) {
```

Comment:
is it not possible to use the one in `kubeutils`? i guess that one doesn't take a kubeconfig

Reply:
i would have to move all this to use the go client